lib/services/permission_service.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:async';
+import 'dart:io' show Directory;
+import 'package:permission_handler/permission_handler.dart';
+
+import '../config.dart';
+
+class PermissionService {
+  final PermissionHandler _permissionHandler = PermissionHandler();
+  final List<PermissionGroup> _requiredPermissions = [
+    PermissionGroup.microphone,
+    PermissionGroup.storage
+  ];
+
+
+  Future<bool> hasPermission(PermissionGroup permission) async {
+    var permissionStatus =
+    await _permissionHandler.checkPermissionStatus(permission);
+    return permissionStatus == PermissionStatus.granted;
+  }
+
+
+  Future<bool> _requestPermission(List<PermissionGroup> permissions) async {
+    var result = await _permissionHandler.requestPermissions(permissions);
+    bool granted = true;
+    result.forEach((permission, permissionStatus) {
+      if (permissionStatus != PermissionStatus.granted)
+        granted = granted && false;
```

Comment:
how about we get rid of the if statement  ```granted = granted && permissionStatus == PermissionStatus.granted```

Reply:
This is actually elegant! Thanks